src/types.rs

```diff
@@ -0,0 +1,147 @@
+use pyo3::{
+    prelude::*,
+    types::{PyDict, PyList},
+    PyNativeType, ToBorrowedObject, ToPyObject,
+};
+
+pub trait PyArray<'py>: Into<PyObject> + Clone + 'py {
+    fn new<T, U>(py: Python<'py>, elements: impl IntoIterator<Item = T, IntoIter = U>) -> Self
+    where
+        T: ToPyObject,
+        U: ExactSizeIterator<Item = T>;
```

Comment:
If you replace this `new` with 

```
fn pythonize_sequence(py: Python<'py>, elements: Vec<PyObject>) -> PyObject
```

that's probably good enough for now for the to-python direction, and the from-python direction can be handled by the `PySequence` type in PyO3.

I don't think this trait should need anything else.